src/rpdk/core/contract/resource_client.py

```diff
@@ -413,20 +421,39 @@ def _call(self, payload):
             json.dumps(payload_to_log, ensure_ascii=False, indent=2),
         )
         payload = json.dumps(payload, ensure_ascii=False, indent=2)
-        result = self._client.invoke(
-            FunctionName=self._function_name, Payload=payload.encode("utf-8")
-        )
-
-        try:
-            payload = json.load(result["Payload"])
-        except json.decoder.JSONDecodeError as json_error:
-            LOG.debug("Received invalid response\n%s", result["Payload"])
-            raise ValueError(
-                "Handler Output is not a valid JSON document"
-            ) from json_error
+        if self._docker_image:
+            if not self._executable_entrypoint:
+                raise InvalidProjectError(
+                    "executableEntrypoint not set in .rpdk-config. "
+                    "Have you run cfn generate?"
+                )
+            result = (
+                self._docker_client.containers.run(
+                    self._docker_image,
+                    self._executable_entrypoint + " '" + payload + "'",
+                )
+                .decode()
+                .strip()
+            )
+            LOG.debug("=== Handler execution logs ===")
+            LOG.debug(result)
+            # pylint: disable=W1401
+            regex = "StartResponse-([\s\S]*)-EndResponse"  # noqa: W605
```

Comment:
I'm afraid  `StartResponse-` and `-EndResponse` is not special enough,  it's possible customer may have property value like this.. Think about something like `__CFN_RESOURCE_START_RESPONSE__`

Reply:
Changed